src/file.ts

```diff
@@ -3560,15 +3564,31 @@ class File extends ServiceObject<File> {
       typeof optionsOrCallback === 'function' ? optionsOrCallback : callback;
     const options =
       typeof optionsOrCallback === 'object' ? optionsOrCallback : {};
-
-    const writable = this.createWriteStream(options)
-      .on('error', callback!)
-      .on('finish', callback!);
-    if (options.onUploadProgress) {
-      writable.on('progress', options.onUploadProgress);
+    const isMultipart = options.resumable === false;
+    const returnValue = retry(async (bail: (err: Error) => void) => {
```

Comment:
This is a future todo: How can retry config be uniform between this retry logic versus what's in nodejs-common repo?

Reply:
Which logic in `nodejs-common` are you referring to here?

Reply:
I was referring to [autoRetry and maxRetries](https://github.com/googleapis/nodejs-common/blob/611d16ba465b53d5eeb3ad59d37a2501a77c3b87/src/util.ts#L118-L134) options.

Reply:
If I understand correctly, this logic is retrying certain requests. What we're doing here is retrying all the logic, not just the request. I believe that in order to do something similar, we'll have to refactor `File.save` so it passes `data` to `File.createWriteStream`. Then, on request failure, we can do a similar retry.